test/integration/nodeagent/lease/lease_test.go

```diff
@@ -0,0 +1,158 @@
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+
+	leasecontroller "github.com/gardener/gardener/pkg/nodeagent/controller/lease"
+	gardenerutils "github.com/gardener/gardener/pkg/utils"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+)
+
+var _ = Describe("Reconcile", func() {
+
+	Describe("Lease controller tests", func() {
+		var (
+			node      *corev1.Node
+			nodeName  string
+			fakeclock *testing.FakeClock
+		)
+
+		BeforeEach(func() {
+			nodeName = "test-" + gardenerutils.ComputeSHA256Hex([]byte(uuid.NewUUID()))[:8]
+			fakeclock = testing.NewFakeClock(time.Now().UTC())
+
+			By("Setup manager")
+			mgr, err := manager.New(restConfig, manager.Options{
+				Metrics: metricsserver.Options{BindAddress: "0"},
+				Cache: cache.Options{
+					DefaultNamespaces: map[string]cache.Config{testNamespace.Name: {}},
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Register controller")
+			leaseReconciler := &leasecontroller.Reconciler{
+				Clock:                fakeclock,
+				Namespace:            testNamespace.Name,
+				HostName:             nodeName,
+				RenewIntervalSeconds: 3,
+			}
+			Expect(leaseReconciler.AddToManager(mgr)).To(Succeed())
+
+			By("Start manager")
+			mgrContext, mgrCancel := context.WithCancel(ctx)
+
+			go func() {
+				defer GinkgoRecover()
+				Expect(mgr.Start(mgrContext)).To(Succeed())
+			}()
+
+			DeferCleanup(func() {
+				By("Stop manager")
+				mgrCancel()
+			})
+
+			node = &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   nodeName,
+					Labels: map[string]string{testID: testRunID},
+				},
+			}
+
+			By("Create Node")
+			Expect(testClient.Create(ctx, node)).To(Succeed())
+			DeferCleanup(func() {
+				By("Delete Node")
+				Expect(client.IgnoreNotFound(testClient.Delete(ctx, node))).To(Succeed())
+				By("Delete Lease")
+				lease := &coordinationv1.Lease{ObjectMeta: metav1.ObjectMeta{Namespace: testNamespace.Name, Name: "gardener-node-agent-" + nodeName}}
+				Expect(client.IgnoreNotFound(testClient.Delete(ctx, lease))).To(Succeed())
+			})
+		})
+
+		It("Reconcile should create lease", func() {
+			lease := &coordinationv1.Lease{}
+			Eventually(func() error {
+				return testClient.Get(ctx, types.NamespacedName{Namespace: testNamespace.Name, Name: "gardener-node-agent-" + nodeName}, lease)
+			}).Should(Succeed())
+			validateOwnerReference(lease, node)
+		})
+
+		It("Reconcile should update lease time", func() {
+			lease := &coordinationv1.Lease{}
+			Eventually(func() error {
+				return testClient.Get(ctx, types.NamespacedName{Namespace: testNamespace.Name, Name: "gardener-node-agent-" + nodeName}, lease)
+			}).Should(Succeed())
+			validateOwnerReference(lease, node)
+			oldRenewTime := lease.Spec.RenewTime
+			fakeclock.SetTime(fakeclock.Now().Add(1 * time.Minute))
```

Comment:
Why `1m`? Shouldn't we use something closer to the configured `RenewIntervalSeconds` to make the assertion below more accurate?